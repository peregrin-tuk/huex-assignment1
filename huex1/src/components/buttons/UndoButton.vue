<template>
  <IconButton
    :icon="icon"
    class="undo-button"
    @click-event="$store.dispatch('undoHistory')"
  />
</template>

<script lang="ts">
import Vue from "vue";
import IconButton from "./IconButton.vue"

export default Vue.extend({
  name: "ToolButton",
  components: {
    IconButton
  },
  data: () => ({
    icon: 'mdi-undo'
  }),
});
</script>

<style lang="scss" scoped>
</style>
