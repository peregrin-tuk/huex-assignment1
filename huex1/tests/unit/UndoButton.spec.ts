import {createLocalVue, shallowMount, mount} from '@vue/test-utils';
import UndoButton from "@/components/buttons/UndoButton.vue";
// eslint-disable-next-line no-unused-vars
import Vuex, { Store } from 'vuex'
import Vuetify from 'vuetify';

const localVue = createLocalVue()
localVue.use(Vuex)
localVue.use(Vuetify)

describe('BoardTitle', () => {
  let actions: { undoHistory: jest.Mock};
  let store: Store<any>;

  beforeEach(() => {
    actions = {
      undoHistory: jest.fn()
    }
    store = new Vuex.Store({
      actions,
      state: {
        currentBoard: {
          name: 'Awesome name',
          key: '#123456789',
          content: [
            { name: 'first shape' },
            { name: 'second shape' }
          ],
          id: '1234'
        }
      }
    })
  })

  it('should match snapshot', () => {
    const wrapper = shallowMount(UndoButton, { 
      store,
      localVue,
    });

    expect(wrapper.html()).toMatchSnapshot();
  })

  it('should remove latest element from current board content when clicked', () => {
    const wrapper = mount(UndoButton, { 
      store, 
      localVue,
    });

    const button = wrapper.find('.v-btn.undo-button');

    button.trigger('click');

    expect(actions.undoHistory).toHaveBeenCalledTimes(1);
  })
})
