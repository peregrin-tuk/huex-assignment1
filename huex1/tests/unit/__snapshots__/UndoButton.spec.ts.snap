// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BoardTitle should match snapshot 1`] = `<iconbutton-stub icon="mdi-undo" class="undo-button"></iconbutton-stub>`;
